LongWarOfTheChosen/Src/LW_Overhaul/Classes/LWTemplateMods.uc

```diff
@@ -1967,13 +1969,13 @@ function SwapExplosiveFalloffItem(X2ItemTemplate Template, int Difficulty)
 		FalloffDamageEffect.UnitDamageSteps=default.UnitDamageSteps;
 		FalloffDamageEffect.EnvironmentDamageSteps=default.EnvironmentDamageSteps;
 
-		if (ThrownDamageEffect != none)
+		if (ThrownDamageEffect != none && ValidExplosiveFalloffItem(Template, ThrownDamageEffect, LaunchedDamageEffect))
```

Comment:
> 
> 
> Do you need to do this check here? Execution shouldn't even reach this point if `ValidExplosiveFalloffItem()` returns `false`.

I think what I was doing was trying my best to copy whatever SwapExplosiveFalloffAbility did, and do so without bricking the game. Does the check being there do anything bad?

Reply:
It doesn't do anything bad per se, but it's confusing and makes the code a bit harder to understand. The reason the code seems to check twice for `ThrownDamageEffect` and `LaunchedDamageEffect` is because the parent condition checks whether *either* of them are `none`, so one of them may still be `none` inside the condition. But the check you added is an *AND*, so it's guaranteed to be `true` inside the condition.

Anyway, thanks for the clarification. I'll remove the unnecessary code and merge.